indica/readers/abstractreader.py

```diff
@@ -544,20 +519,33 @@ def get_equilibrium(
             A dictionary containing the requested physical quantities.
 
         """
-        location_quantities = {"psi", "rmag", "zmag", "faxs", "fbnd"}
+        global_quantities = {
+            "psi",
+            "rmag",
+            "zmag",
+            "rgeo",
+            "faxs",
+            "fbnd",
+            "ipla",
+            "wp",
+            "df",
+        }
         separatrix_quantities = {"rbnd", "zbnd"}
         flux_quantities = {"f", "ftor", "vjac", "rmji", "rmjo"}
-        available_quantities = self.available_quantities(calculation)
+        available_quantities = self.available_quantities(instrument)
         if len({"rmji", "rmjo"} & quantities) > 0:
             quantities.add("zmag")
         if "faxs" in quantities:
             quantities |= {"rmag", "zmag"}
-        database_results = self._get_equilibrium(uid, calculation, revision, quantities)
+        database_results = self._get_equilibrium(uid, instrument, revision, quantities)
+        if len(database_results) == 0:
+            print(f"No data from {uid}.{instrument}:{revision}")
+            return database_results
+        _revision = database_results["revision"]
+
         diagnostic_coord = "rho_poloidal"
         times = database_results["times"]
-        downsample_ratio = int(
-            np.ceil((len(times) - 1) / (times[-1] - times[0]) / self._max_freq)
-        )
+        times_unique, ind_unique = np.unique(times, return_index=True)
```

Comment:
Should probably keep the `downsample_ratio` feature and add the unique time check as well, rather than instead of